src/librustc_codegen_llvm/type_of.rs

```diff
@@ -60,9 +59,7 @@ fn uncached_llvm_type<'a, 'tcx>(
         // ty::Dynamic(..) |
         ty::Foreign(..) |
         ty::Str => {
-            let mut name = String::with_capacity(32);
-            let printer = DefPathBasedNames::new(cx.tcx, true, true);
-            printer.push_type_name(layout.ty, &mut name, false);
+            let mut name = format!("{}", layout.ty);
```

Comment:
This can be written as `layout.ty.to_string()`.

Reply:
Fixed!